src/wp-includes/functions.php

```diff
@@ -8061,6 +8061,91 @@ function wp_unique_id_from_values( array $data, string $prefix = '' ): string {
 	return $prefix . $hash;
 }
 
+/**
+ * Retrieves cached query data if valid and unchanged.
+ *
+ * @param string $cache_key The cache key used for storage and retrieval.
+ * @param string $group The cache group used for organizing data.
+ * @param string $last_changed The timestamp of the last modification to the cache group.
+ * @return mixed|false The cached data if valid, or false if the cache does not exist or is outdated.
+ */
+function wp_cache_get_query_data( $cache_key, $group, $last_changed ) {
```

Comment:
Do we want to add these methods to the `object-cache.php` instead, to the cache implementations can adopt these?

Reply:
I think it would be helpful to place these in object-cache & cache-compat to allow for implementers to adapt as needed.

I discussed passing the last changed groups to the functions with Till and we think that would be more helpful. I know this will result in more calls to `wp_cache_get( 'last_changed' ... )` but I don't think that's a big problem as they will be in the local, PHP, memory on the second and subsequent calls rather than need to connect to the persistent caching server each time.